app/assets/javascripts/deal_poller.js

```diff
@@ -0,0 +1,54 @@
+function DealPoller(options){
+  this.$liveDealsTimerElement = options.$liveDealsTimerElement;
+  this.$modalElement = options.$modalElement;
+  this.pollingInterval = options.pollingInterval;
+}
+
+DealPoller.prototype.init = function(){
+  this.setPolling();
+};
+
+DealPoller.prototype.setPolling = function(){
+  var _this = this;
+  if(this.$liveDealsTimerElement.length > 0)
+    setInterval(_this.pollingForExpiration, this.pollingInterval, _this);
+};
+
+DealPoller.prototype.pollingForExpiration = function(_this){
```

Comment:
check for deal is sellable or not
including quantity check
send a request on a separate action